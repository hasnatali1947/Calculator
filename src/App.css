
* {
  padding: 0;
  margin: 0;
}

body {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
}

nav {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  background-color: white;
}

.navbar-container ul {
  display: flex;
  justify-content: space-around;
  align-items: center;
  border: 2px solid;
  height: 8vh;
  list-style: none;
}

.Math-Mag {
  font-size: 24px;
}

.li {
  display: flex;
  font-size: 3vh;
}

.li li {
  padding: 1rem;
  transition: 0.3s ease-in-out;
}

.li li:hover {
  scale: 0.95;
  transition: 0.3s ease-in-out;
}

.navbar-container a {
  text-decoration: none;
  color: black;
}

.homepage {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 50vh;
  margin-top: -30vh;
  font-size: 21px;
}

.top {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 70%;
  text-align: center;
  padding: 20px 0;
}

.top h1 {
  padding: 20px 0;
  font-size: 42px;
}

.bottom {
  width: 70%;
  text-align: center;
  padding-top: 20px;
}

.calculatorPage {
  display: flex;
}

.calculator-h2 {
  font-size: 42px;
}

.row {
  display: flex;
  flex-direction: row;
  align-items: stretch;
  height: 100%;
}

.Calc-container {
  margin-top: 10vh;
  border: 0.5px solid;
  width: 70vh;
  display: flex;
  flex-direction: column;
}

.input {
  display: flex;
  align-items: center;
  justify-content: end;
  height: 10vh;
  background-color: rgb(161, 153, 153);
  width: 100%;
  border: none;
  outline: none;
  font-size: 27px;
  text-indent: 5px;
  color: #fff;
}

.input::-webkit-outer-spin-button,
.input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
  padding-right: 5px;
}

.row input {
  height: 10vh;
  flex: 1;
  border: 0.5px solid rgb(149, 149, 149);
  color: #000;
  font-weight: 600;
  cursor: pointer;
}

.row .button-0 {
  flex: 2;
}

.row .colourButton {
  background-color: rgba(251, 139, 12, 0.894);
}

.Jokes-Section {
  margin-top: 20px;
  display: flex;
  align-items: center;
  flex-direction: column;
}

.jokes {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 15vh;
  width: 80vh;
  border: 2px solid;
  padding: 4px;
  border-radius: 10px;
}

.jokesButton {
  height: 6vh;
  width: 17vh;
  background-color: rgb(38, 233, 233);
  margin-top: 20px;
  border: 2px solid rgb(38, 233, 233);
  border-radius: 7px;
  cursor: pointer;
  font-size: 14px;
  font-weight: 500;
  transition: 0.3s ease-in-out;
}

.jokesButton:hover {
  scale: 0.96;
}
