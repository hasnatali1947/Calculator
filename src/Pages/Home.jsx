import React from 'react';

export default function Home() {
  return (
    <section className="homepage">
      <div className="top">
        <h1>Welcome to our page</h1>
        <p>
          Welcome to our website, where you can find a simple calculator
          to make your calculations easy and quick. Our calculator has a
          user-friendly interface that is easy to use, even if you&apos;re not
          a math expert. It can perform basic arithmetic operations like
          addition, subtraction, multiplication, and division, as well as
          more complex calculations like percentages, square roots, and
          exponents.
        </p>
      </div>
      <div className="bottom">
        <p>
          In addition to our calculator, we also have a collection of inspiring
          quotes to brighten up your day. These quotes cover a variety of topics
          such as motivation, success, happiness, and more. We update our quotes
          regularly, so you can always find fresh inspiration whenever you visit
          our website.
        </p>
      </div>
    </section>
  );
}
