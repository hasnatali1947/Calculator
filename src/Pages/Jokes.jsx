import React from 'react';
import Ninja from '../Hooks/Api';

export default function jokes() {
  return (
    <Ninja />
  );
}
