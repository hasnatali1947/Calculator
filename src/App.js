import './App.css';
import React from 'react';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import Calculator from './Pages/calculatorPage';
import Home from './Pages/Home';
import Jokes from './Pages/Jokes';
import Navbar from './components/navbar';

const App = () => (
  <Router>
    <Navbar />
    <Routes>
      <Route exact path="/" element={<Home />} />
      <Route exact path="/CalculatorPage" element={<Calculator />} />
      <Route exact path="/jokes" element={<Jokes />} />
    </Routes>
  </Router>
);

export default App;
