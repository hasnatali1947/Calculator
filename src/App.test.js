import React from 'react';
import renderer, { create } from 'react-test-renderer';
import App from './App';
import Home from './Pages/Home';
import Jokes from './Pages/Jokes';
import CalculatorPage from './Pages/calculatorPage';
import operate from './logic/operate';
import calculate from './logic/calculate';

describe('Operate Tests', () => {
  test('1 + 2 = 3', () => {
    const result = operate(1, 2, '+');
    expect(result).toBe('3');
  });

  test('0 + 100 = 100', () => {
    const result = operate(0, 100, '+');
    expect(result).toBe('100');
  });

  test('1 - 2 = -1', () => {
    const result = operate(1, 2, '-');
    expect(result).toBe('-1');
  });

  test('10 - 10 = 0', () => {
    const result = operate(10, 10, '-');
    expect(result).toBe('0');
  });

  test('2 x 3 = 6', () => {
    const result = operate(2, 3, 'x');
    expect(result).toBe('6');
  });

  test('0 x 3 = 0', () => {
    const result = operate(0, 3, 'x');
    expect(result).toBe('0');
  });

  test('5 % 5 = 0', () => {
    const result = operate(5, 5, '%');
    expect(result).toBe('0');
  });

  test('3 % 2 = 1', () => {
    const result = operate(3, 2, '%');
    expect(result).toBe('1');
  });

  test('4 ÷ 5 = 0.8', () => {
    const result = operate(4, 5, '÷');
    expect(result).toBe('0.8');
  });

  test('0 ÷ 5 = 0', () => {
    const result = operate(0, 5, '÷');
    expect(result).toBe('0');
  });
});

describe('Calculate Test', () => {
  test('5 + 5 equal 10', () => {
    const object = {
      total: '5',
      next: '5',
      operation: '+',
    };
    const newObj = calculate(object, '');
    expect(newObj.total).toBe('10');
  });

  test('10 - 8 = 0', () => {
    const object = {
      total: '10',
      next: '8',
      operation: '-',
    };
    const newObj = calculate(object, '');
    expect(newObj.total).toBe('2');
  });

  test('3 x 9 = 27', () => {
    const object = {
      total: '3',
      next: '9',
      operation: 'x',
    };
    const newObj = calculate(object, '');
    expect(newObj.total).toBe('27');
  });

  test('12 ÷ 10 = 1.2', () => {
    const object = {
      total: '12',
      next: '10',
      operation: '÷',
    };
    const newObj = calculate(object, '');
    expect(newObj.total).toBe('1.2');
  });

  test('2 % 2 = 0', () => {
    const object = {
      total: '2',
      next: '2',
      operation: '%',
    };
    const newObj = calculate(object, '');
    expect(newObj.total).toBe('0');
  });

  test('equal function', () => {
    const object = {
      total: '2',
      next: '2',
      operation: '+',
    };
    const newObj = calculate(object, '=');
    expect(newObj.total).toBe('4');
  });

  test('AC function', () => {
    const object = {
      total: '2',
      next: '2',
      operation: '+',
    };
    const newObj = calculate(object, 'AC');
    expect(newObj.total).toBe(null);
  });

  it('+/- Function', () => {
    const object = {
      total: '6',
      next: '6',
      operation: '+',
    };
    const newObj = calculate(object, '+/-');
    expect(newObj.next).toBe('-6');
  });
});

describe('Components render Correctly', () => {
  test('App render Correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('App render Correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('jokes render Correctly', () => {
    const tree = renderer.create(<Jokes />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('calculatorPage render Correctly', () => {
    const tree = renderer;
    create(<CalculatorPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
