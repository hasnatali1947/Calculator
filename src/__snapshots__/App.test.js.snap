// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test 1`] = `
Array [
  <nav>
    <div
      className="navbar-container"
    >
      <ul
        className="navbar"
      >
        <div>
          <h1
            className="Math-Mag"
          >
            Math Magicions
          </h1>
        </div>
        <div
          className="li"
        >
          <li>
            <a
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/calculatorPage"
              onClick={[Function]}
            >
              CalculatorPage
            </a>
          </li>
          <li>
            <a
              href="/jokes"
              onClick={[Function]}
            >
              Jokes
            </a>
          </li>
        </div>
      </ul>
    </div>
  </nav>,
  <section
    className="homepage"
  >
    <div
      className="top"
    >
      <h1>
        Welcome to our page
      </h1>
      <p>
        Welcome to our website, where you can find a simple calculator to make your calculations easy and quick. Our calculator has a user-friendly interface that is easy to use, even if you're not a math expert. It can perform basic arithmetic operations like addition, subtraction, multiplication, and division, as well as more complex calculations like percentages, square roots, and exponents.
      </p>
    </div>
    <div
      className="bottom"
    >
      <p>
        In addition to our calculator, we also have a collection of inspiring quotes to brighten up your day. These quotes cover a variety of topics such as motivation, success, happiness, and more. We update our quotes regularly, so you can always find fresh inspiration whenever you visit our website.
      </p>
    </div>
  </section>,
]
`;

exports[`App test 2`] = `
<section
  className="homepage"
>
  <div
    className="top"
  >
    <h1>
      Welcome to our page
    </h1>
    <p>
      Welcome to our website, where you can find a simple calculator to make your calculations easy and quick. Our calculator has a user-friendly interface that is easy to use, even if you're not a math expert. It can perform basic arithmetic operations like addition, subtraction, multiplication, and division, as well as more complex calculations like percentages, square roots, and exponents.
    </p>
  </div>
  <div
    className="bottom"
  >
    <p>
      In addition to our calculator, we also have a collection of inspiring quotes to brighten up your day. These quotes cover a variety of topics such as motivation, success, happiness, and more. We update our quotes regularly, so you can always find fresh inspiration whenever you visit our website.
    </p>
  </div>
</section>
`;

exports[`Components render Correctly App render Correctly 1`] = `
Array [
  <nav>
    <div
      className="navbar-container"
    >
      <ul
        className="navbar"
      >
        <div>
          <h1
            className="Math-Mag"
          >
            Math Magicions
          </h1>
        </div>
        <div
          className="li"
        >
          <li>
            <a
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/calculatorPage"
              onClick={[Function]}
            >
              CalculatorPage
            </a>
          </li>
          <li>
            <a
              href="/jokes"
              onClick={[Function]}
            >
              Jokes
            </a>
          </li>
        </div>
      </ul>
    </div>
  </nav>,
  <section
    className="homepage"
  >
    <div
      className="top"
    >
      <h1>
        Welcome to our page
      </h1>
      <p>
        Welcome to our website, where you can find a simple calculator to make your calculations easy and quick. Our calculator has a user-friendly interface that is easy to use, even if you're not a math expert. It can perform basic arithmetic operations like addition, subtraction, multiplication, and division, as well as more complex calculations like percentages, square roots, and exponents.
      </p>
    </div>
    <div
      className="bottom"
    >
      <p>
        In addition to our calculator, we also have a collection of inspiring quotes to brighten up your day. These quotes cover a variety of topics such as motivation, success, happiness, and more. We update our quotes regularly, so you can always find fresh inspiration whenever you visit our website.
      </p>
    </div>
  </section>,
]
`;

exports[`Components render Correctly App render Correctly 2`] = `
<section
  className="homepage"
>
  <div
    className="top"
  >
    <h1>
      Welcome to our page
    </h1>
    <p>
      Welcome to our website, where you can find a simple calculator to make your calculations easy and quick. Our calculator has a user-friendly interface that is easy to use, even if you're not a math expert. It can perform basic arithmetic operations like addition, subtraction, multiplication, and division, as well as more complex calculations like percentages, square roots, and exponents.
    </p>
  </div>
  <div
    className="bottom"
  >
    <p>
      In addition to our calculator, we also have a collection of inspiring quotes to brighten up your day. These quotes cover a variety of topics such as motivation, success, happiness, and more. We update our quotes regularly, so you can always find fresh inspiration whenever you visit our website.
    </p>
  </div>
</section>
`;

exports[`Components render Correctly calculatorPage render Correctly 1`] = `
Object {
  "_Scheduler": Object {
    "reset": [Function],
    "unstable_IdlePriority": 5,
    "unstable_ImmediatePriority": 1,
    "unstable_LowPriority": 4,
    "unstable_NormalPriority": 3,
    "unstable_Profiling": null,
    "unstable_UserBlockingPriority": 2,
    "unstable_advanceTime": [Function],
    "unstable_cancelCallback": [Function],
    "unstable_clearYields": [Function],
    "unstable_continueExecution": [Function],
    "unstable_flushAll": [Function],
    "unstable_flushAllWithoutAsserting": [Function],
    "unstable_flushExpired": [Function],
    "unstable_flushNumberOfYields": [Function],
    "unstable_flushUntilNextPaint": [Function],
    "unstable_forceFrameRate": [Function],
    "unstable_getCurrentPriorityLevel": [Function],
    "unstable_getFirstCallbackNode": [Function],
    "unstable_next": [Function],
    "unstable_now": [Function],
    "unstable_pauseExecution": [Function],
    "unstable_requestPaint": [Function],
    "unstable_runWithPriority": [Function],
    "unstable_scheduleCallback": [Function],
    "unstable_setDisableYieldValue": [Function],
    "unstable_shouldYield": [Function],
    "unstable_wrapCallback": [Function],
    "unstable_yieldValue": [Function],
  },
  "act": [Function],
  "create": [Function],
  "unstable_batchedUpdates": [Function],
}
`;

exports[`Components render Correctly jokes render Correctly 1`] = `"something went wrong!"`;

exports[`calculatorPage Test 1`] = `
Object {
  "_Scheduler": Object {
    "reset": [Function],
    "unstable_IdlePriority": 5,
    "unstable_ImmediatePriority": 1,
    "unstable_LowPriority": 4,
    "unstable_NormalPriority": 3,
    "unstable_Profiling": null,
    "unstable_UserBlockingPriority": 2,
    "unstable_advanceTime": [Function],
    "unstable_cancelCallback": [Function],
    "unstable_clearYields": [Function],
    "unstable_continueExecution": [Function],
    "unstable_flushAll": [Function],
    "unstable_flushAllWithoutAsserting": [Function],
    "unstable_flushExpired": [Function],
    "unstable_flushNumberOfYields": [Function],
    "unstable_flushUntilNextPaint": [Function],
    "unstable_forceFrameRate": [Function],
    "unstable_getCurrentPriorityLevel": [Function],
    "unstable_getFirstCallbackNode": [Function],
    "unstable_next": [Function],
    "unstable_now": [Function],
    "unstable_pauseExecution": [Function],
    "unstable_requestPaint": [Function],
    "unstable_runWithPriority": [Function],
    "unstable_scheduleCallback": [Function],
    "unstable_setDisableYieldValue": [Function],
    "unstable_shouldYield": [Function],
    "unstable_wrapCallback": [Function],
    "unstable_yieldValue": [Function],
  },
  "act": [Function],
  "create": [Function],
  "unstable_batchedUpdates": [Function],
}
`;

exports[`jokes test 1`] = `"something went wrong!"`;
